### ISSUE TITLE
Loading a sprite fails when using absolute path on apportable

### ISSUE THREAD

**Issue by agro1986** [cocos2d/cocos2d-objc#857]
Suppose we want to load a sprite from the user's documents folder. First we need to get the absolute path of the image file.

But if we pass the absolute path to [CCSprite spriteWithImageNamed:], it will fail on Apportable because of this issue of Apportable's implementation of NSString's stringByAppendingPathComponent:
https://code.google.com/p/apportable/issues/detail?id=809&colspec=ID%20Type%20Status%20Milestone%20Owner%20Summary%20BlockedOn%20Component%20FoundIn%20FixedIn

The temporary solution is to edit CCFileUtils.m, -(NSString_) fullPathForFilename:(NSString_)filename contentScale:(CGFloat *)contentScale, and replace this line:

```
NSString *fileWithPath = [path stringByAppendingPathComponent:newfilename];
```

With this:

```
#ifdef ANDROID
            // to handle issue #809 of Apportable
            NSString *fileWithPath;
            if( path.length == 0) {
                fileWithPath = newfilename;
            } else {
                fileWithPath = [path stringByAppendingPathComponent:newfilename];
            }
#else            
            NSString *fileWithPath = [path stringByAppendingPathComponent:newfilename];
#endif
```

In my experience, Apportable is kind of slow in fixing bugs, so I suggest cocos2d being proactive and put this line until Apportable fixed it.

Thanks

**Comment by paulb777** [comment]
The stringByAppendingPathComponent bug has been fixed in Apportable master and will be available in the next SDK release (planned to be 1.1.14)

**Comment by s1ddok** [comment]
Closed due to Apportable death.
